### PR TITLE
ssl_tls12_server: fix potential NULL-dereferencing if local certifica…

### PR DESCRIPTION
**Pull request by leorosen**
## Description
fix potential NULL-dereferencing if local certificate was not set.

See #6183 

## Status
READY

## Requires Backporting
 NO  

## Migrations
NO

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
Ignore the CI failure, that was a random infrastructure glitch and the same job succeeded on -head.

**Review by gilles-peskine-arm — DISMISSED** [review]
This is a robustness improvement, so approved without a changelog entry or a test.

File: library/ssl_tls12_server.c
Comment (daverodgman) [inline library/ssl_tls12_server.c:3410]: Should have spaces around brackets

File: library/ssl_tls12_server.c
Comment (daverodgman) [inline library/ssl_tls12_server.c:3407]: We normally always initialise return values to MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED in this manner - so I don't see any reason to change this.

**Review by daverodgman — CHANGES_REQUESTED** [review]

File: library/ssl_tls12_server.c
Comment (leorosen) [reply library/ssl_tls12_server.c:3407]: This change is in response to a separate Coverity warning "value not used", which rightfully indicates that the variable will be assigned a different value before it is read from the first time. Normally I would just ignore such a low level warning, but since I was touching this function anyway, I decide to address this one too.

File: library/ssl_tls12_server.c
Comment (leorosen) [reply library/ssl_tls12_server.c:3410]: Will fix.

File: library/ssl_tls12_server.c
Comment (gilles-peskine-arm) [reply library/ssl_tls12_server.c:3407]: Initializing variables to a constant regardless of how they're used later is very common practice. Some coding rules even forbid uninitialized variables (ours doesn't: we think that the best practice depends on how the variable is used.)

This particular idiom with `ret` and `status` variables is used throughout Mbed TLS. Is your Coverity instance complaining about all of them? (Ours isn't.)

**Review by daverodgman — CHANGES_REQUESTED** [review]
Please could you sort out the removal of MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED - thanks

**Comment by daverodgman** [comment]
It maybe makes sense to backport this? Trivial backport in #6215 . @gilles-peskine-arm feel free to close the backport if you disagree.

**Review by gilles-peskine-arm — APPROVED** [review]

**Comment by leorosen** [comment]
PR approved.

**Comment by daverodgman** [comment]
> PR approved.

It's approved, but not yet merged! Please leave the PR open after approval - it will usually get merged soon afterwards (sometimes there is a wait for CI to complete, or backports to be ready, etc). It will be closed when it gets merged.

Thanks for your contributions on this one.

**Comment by leorosen** [comment]
> > PR approved.
> 
> It's approved, but not yet merged! Please leave the PR open after approval - it will usually get merged soon afterwards (sometimes there is a wait for CI to complete, or backports to be ready, etc). It will be closed when it gets merged.
> 
> Thanks for your contributions on this one.

Thank you David, I was looking for a "Merge" button, and clocked "Close" in error.

**Pull request merged**
